src/components/Correlation/index.js

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { uuid } from "../../utils";
+
+const createCorrelation = () => {
+  return {
+    lifecycle: {
+      onBeforeEvent(event, options) {
+        return Promise.resolve(options.correlationID).then(correlationID => {
+          if (correlationID !== undefined) {
+            event.correlationID = correlationID;
+          }
+        });
+      }
+    },
+    commands: {
+      createCorrelationID: uuid
```

Comment:
I think we're going to need a `key` or `consumer` of each correlation ID. For example, a customer can generate multiple correlation IDs on the same page/view for different flows, and might want to get them by `key` and share them with 3rd party vendors later on in the page lifecycle:

```
const loginCorrelationID = correlation.createCorrelationID("login");
const checkoutCorrelationID = correlation.createCorrelationID("checkout");
....
....
const checkoutCorrelationID = correlation.getCorrelationID("checkout");
```

or maybe `createCorrelationID` creates an ID if it doesn't exist, or creates a new one and returns.

Let's confirm with Daniel and Lubby.

Reply:
Yeah, that's interesting. You're thinking they'll want to retrieve them by a key because they won't want to pass the correlation ID around?

Would you like me to wait until we figure this out before merging?

Reply:
Yeah let's wait to learn more. I will start the conversation with Daniel and team as soon as possible and include you.